### ISSUE TITLE
cached key not getting deleted

### ISSUE THREAD

**Issue by sachins-geekyants** [atsign-foundation/at_client_sdk#908]
### Describe the bug

When an atKey is deleted from the sender's side, the cached copy is not getting deleted from receiver's end.

AtKey :
```
 var key = AtKey()
      ..key = "test_delete"
      ..sharedBy = AtClientManager.getInstance().atClient.getCurrentAtSign()
      ..sharedWith = "@snowboard89"
      ..metadata = Metadata()
      ..metadata!.ccd = true
      ..metadata!.ttr = -1;
      
      await AtClientManager.getInstance().atClient.put(key, "value");
      
      AtClientManager.getInstance().atClient.delete(key);
```

On receiver's end:
`cached:@snowboard89:test_delete.atbuzz@local2010`

### Steps to reproduce

1. Share an AtKey like mentioned above.
2. share with an atSign and then delete.
3. lookup for the key on receiver's end

### Expected behavior

The cached key should get deleted.

### Smartphones

- iPhone 12 simulator

### Were you using an atApplication when the bug was found?

atBuzz

**Comment by Sardar1208** [comment]
@sitaram-kalluri Do you have any updates on this ?

**Comment by sitaram-kalluri** [comment]
> @sitaram-kalluri Do you have any updates on this ?

The issue is fixed and changes are available in following branch: [fix_cached_key_deletion](https://github.com/atsign-foundation/at_client_sdk/tree/fix_cached_key_deletion)

**Comment by sitaram-kalluri** [comment]
The changes are published in at_client v3.0.55
